src/Options/Factory.hpp

```diff
@@ -63,7 +68,14 @@ std::string help_derived() noexcept {
 // It should be inlined once everything is converted.
 template <typename BaseClass, typename Metavariables, typename = std::void_t<>>
 struct get_creatable_classes {
-  using type = tmpl::at<typename Metavariables::factory_classes, BaseClass>;
+  using factory_creation = typename Metavariables::factory_creation;
+#ifdef SPECTRE_DEBUG
+  // This assertion is normally done in tests, but the executable
+  // metavariables don't have tests, so we do it here.
+  using corforming_assertion = std::bool_constant<tt::assert_conforms_to<
+      factory_creation, Options::protocols::FactoryCreation>>;
+#endif  // SPECTRE_DEBUG
```

Comment:
I think it's just fine to assert protocol conformance in consuming code like this:

```suggestion
static_assert(tt::assert_conforms_to<factory_creation, Options::protocols::FactoryCreation>);
```

Conforming types should also assert this in their unit tests so they don't rely on the consuming code to test the protocol conformance "in production" (though as you say, metavars currently don't have unit tests).

Reply:
> Conforming types should also assert this in their unit tests so they don't rely on the consuming code to test the protocol conformance "in production" (though as you say, metavars currently don't have unit tests).

Are you asking me to change anything here?

Reply:
No, I'm only suggesting the diff above :)